clients/src/main/java/org/apache/kafka/common/record/RecordsUtil.java

```diff
@@ -89,10 +89,11 @@ protected static ConvertedRecords<MemoryRecords> downConvert(Iterable<? extends
                 buffer = Utils.ensureCapacity(buffer, buffer.position() + recordBatchAndRecords.batch.sizeInBytes());
                 recordBatchAndRecords.batch.writeTo(buffer);
             } else {
-                MemoryRecordsBuilder builder = convertRecordBatch(toMagic, buffer, recordBatchAndRecords);
-                buffer = builder.buffer();
-                temporaryMemoryBytes += builder.uncompressedBytesWritten();
-                numRecordsConverted += builder.numRecords();
+                try (MemoryRecordsBuilder builder = convertRecordBatch(toMagic, buffer, recordBatchAndRecords)) {
```

Comment:
Do we really need this close? the builder returned by `convertRecordBatch` is already closed. see

https://github.com/apache/kafka/blob/d0e516a87207b04f6266c59712775b5f01c1ed87/clients/src/main/java/org/apache/kafka/common/record/RecordsUtil.java#L124

Additionally, `RecordsUtil` is useless in trunk and it will be removed by #18897

Reply:
> I'll go ahead and close the PR.

yes and thanks for your contribution. Please feel free to ping me if you have other PRs needs to be reviewed.